core/helper/view_utils.py

```diff
@@ -241,44 +281,46 @@ def get_context_data(self, **kwargs):
     def get_search_results_context(self, context):
         return context[self.context_object_name]
 
-    @staticmethod
-    def send_csv_email(csv_handler, queryset, to_email):
-        csv_path = file_utils.create_new_tmp_file_path()
-        csv_handler.create_csv_file(csv_path, queryset)
-
-        if not to_email:
-            log.error(f'unknown user email -- [{to_email}]')
-
-        resp = email_utils.send_email(
-            to_email,
-            subject='Search result',
-            attachments=[
-                ('search_result.csv', open(csv_path, mode='rb'), 'text/csv')
-            ],
-        )
-        log.info(f'csv file email have be send to [{to_email}]')
-        os.remove(csv_path)
-        log.debug('email resp', resp)
-
-    def resp_download_csv(self, request, *args, **kwargs):
+    def resp_file_download(self, request,
+                           file_fn: Callable[[], str],
+                           *args, **kwargs):
 
         def _fn():
             try:
-                self.send_csv_email(self.download_csv_handler, self.get_queryset(), request.user.email)
+                log.debug(f'start send email[{request.user}]....')
+                file_name = file_fn()
+                send_email_file_by_url(file_name, request.user.email)
             except Exception as e:
-                log.error('send csv email fail....')
+                log.error('send email fail....')
                 log.exception(e)
 
-        # create csv file and send email in other process
-        Process(target=_fn).start()
+        # create file and send email in other thread
+        Thread(target=_fn).start()
 
         # stay as same page
-        self.to_user_messages = ['Csv file will be send to your email later.']
+        self.to_user_messages = ['File will be send to your email later.']
```

Comment:
"File will be send to your email later." => "The selected data is being processed and will be sent to your email soon."